webpack.common.js

```diff
@@ -12,75 +12,85 @@ const { CleanWebpackPlugin } = require('clean-webpack-plugin');
 
-module.exports = {
-  entry: {
-    mapbox: './node_modules/mapbox-gl/dist/mapbox-gl.js',
-    main: ['@babel/polyfill', path.resolve(__dirname, 'public/main.js')]
-  },
-  mode: 'development',
-  output: {
-    path: path.resolve(__dirname, 'build/release'),
-    filename: '[name].bundle.js'
-  },
-  module: {
-    noParse: /iconv-loader\.js/,
-    rules: [
-      {
-        test: /\.(png|jpg|gif|svg)$/,
-        use: ['file-loader']
-      },
-      {
-        test: /\.css$/,
-        use: ['style-loader', 'css-loader'],
-      },
-      {
-        test: /\.scss$/,
-        use: ['style-loader', 'css-loader', 'resolve-url-loader', 'postcss-loader', 'sass-loader'],
-        exclude: /node_modules/
-      },
-      {
-        test: /\.js$/,
-        exclude: /node_modules/,
-        use: {
-          loader: 'babel-loader'
+module.exports = env => {
+  return {
+    entry: {
+      mapbox: './node_modules/mapbox-gl/dist/mapbox-gl.js',
+      main: ['@babel/polyfill', path.resolve(__dirname, 'public/main.js')]
+    },
+    mode: 'development',
+    output: {
+      path: path.resolve(__dirname, 'build/release'),
+      filename: '[name].bundle.js'
+    },
+    module: {
+      noParse: /iconv-loader\.js/,
+      rules: [
+        {
+          test: /\.(png|jpg|gif|svg)$/,
+          use: ['file-loader']
+        },
+        {
+          test: /\.css$/,
+          use: ['style-loader', 'css-loader'],
+        },
+        {
+          test: /\.scss$/,
+          use: ['style-loader', 'css-loader', 'resolve-url-loader', 'postcss-loader', 'sass-loader'],
+          exclude: /node_modules/
+        },
+        {
+          test: /\.js$/,
+          exclude: /node_modules/,
+          use: {
+            loader: 'babel-loader'
+          }
+        },
+        {
+          test: /\.hbs$/,
+          use: {
+            loader: 'handlebars-loader'
+          }
         }
+      ]
+    },
+    externals: {
+      'config': JSON.stringify(require(path.resolve(__dirname, 'public', 'config.json')))
+    },
+    resolve: {
+      alias: {
+
       }
-    ]
-  },
-  externals: {
-    'config': JSON.stringify(require(path.resolve(__dirname, 'public', 'config.json')))
-  },
-  resolve: {
-    alias: {
-    }
-  },
-  optimization: {
-    minimizer: [
-      new TerserPlugin({
-        exclude: /node_modules\/(?!mapbox-gl\/dist)/
-      })
+    },
+    optimization: {
+      minimizer: [
+        new TerserPlugin({
+          exclude: /node_modules\/(?!mapbox-gl\/dist)/
+        })
+      ],
+      occurrenceOrder: true,
+      splitChunks: {
+        chunks: 'all'
+      },
+    },
+    plugins: [
+      new CleanWebpackPlugin(),
+      new CopyWebpackPlugin(['./public/config.json']),
+      new WebappWebpackPlugin('@elastic/eui/lib/components/icon/assets/app_ems.svg'),
+      new HTMLWebpackPlugin({
+        template: 'public/index.hbs',
+        hash: true,
+        httpsRedirect: env && env.httpOnly ? false : true,
```

Comment:
```suggestion
        httpsRedirect: 'httpOnly' in process.env ? false : true,
```

What about chocking for `httpOnly` in `process.env`? that way the variable is available always (I think), at least you can run `httpOnly=YES yarn build` or any other command.

Reply:
Hmm, yes. That seems more intuitive than the webpack `env` variable. Thanks.